platforms/jvm/plugins-java-base/src/main/java/org/gradle/api/plugins/internal/DefaultJavaPluginExtension.java

```diff
@@ -324,6 +327,11 @@ public void consistentResolution(Action<? super JavaResolutionConsistency> actio
         action.execute(project.getObjects().newInstance(DefaultJavaPluginExtension.DefaultJavaResolutionConsistency.class, components, sourceSets, configurations));
     }
 
+    @Override
+    public Property<Boolean> getEnablePreview() {
+        return enablePreview;
+    }
```

Comment:
:x: This should be declared as an abstract property. I don't think you even need to mention it in the implementation class, just let the interface declare it and you should be good.

Reply:
`DefaultJavaPluginExtension` is not an abstract class, and changing it to abstract is a backward-incompatible change.

Reply:
Let's make this abstract then, shall we?

Reply:
Changing the Java extension was moved out of scope of this PR